cmd/sriov/main.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"runtime"
+
+	"github.com/containernetworking/cni/pkg/ipam"
+	"github.com/containernetworking/cni/pkg/ns"
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/intel/sriov-cni/pkg/config"
+	"github.com/intel/sriov-cni/pkg/sriov"
+	"github.com/intel/sriov-cni/pkg/utils"
+	"github.com/vishvananda/netlink"
+)
+
+func init() {
+	// this ensures that main runs only on main thread (thread group leader).
+	// since namespace ops (unshare, setns) are done for a single thread, we
+	// must ensure that the goroutine does not jump from OS thread to thread
+	runtime.LockOSThread()
+}
+
+func cmdAdd(args *skel.CmdArgs) error {
+	netConf, err := config.LoadConf(args.StdinData)
+	if err != nil {
+		return fmt.Errorf("SRIOV-CNI failed to load netconf: %v", err)
+	}
+
+	netns, err := ns.GetNS(args.Netns)
+	if err != nil {
+		return fmt.Errorf("failed to open netns %q: %v", netns, err)
+	}
+	defer netns.Close()
+
+	sm := sriov.NewSriovManager()
+	if err := sm.ApplyVFConfig(netConf); err != nil {
```

Comment:
Do we need apply VF configs when DPDK Mode is enabled? or maybe ApplyVFConfig would fail due to VF is attached with VFIO/UIO driver.

Reply:
This is the main feature of this whole change :) We should be able to apply VF config regardless of what driver the VF is attached to(I've tested with vfio-pci/igb_uio driver and it works). This allows us to apply VF vlan, mac, rate-limit etc. For now we support vlan only.